src/StorageManager/rule.py

```diff
@@ -0,0 +1,294 @@
+                owner = default_owner
+
+            if owner not in nodes_by_owner:
+                nodes_by_owner[owner] = []
+            nodes_by_owner[owner].append(node)
+
+        self.nodes_by_owner = nodes_by_owner
+
+    def send_emails_to_owners(self):
+        for owner, nodes in self.nodes_by_owner.items():
+            self.send_email_to_owner(owner, nodes)
+
+    def send_email_to_owner(self, owner, nodes):
+        preview, report = self.generate_report(owner, nodes)
+        recipients = owner.split(",")
+        cc = self.smtp["cc"].split(",")
+        subject = self.generate_subject(owner, nodes, preview, report)
+        content = self.generate_content(owner, nodes, preview, report)
+        send_email(self.smtp, recipients, cc, subject, content, [report])
+
+    def generate_report(self, owner, nodes, preview_len=20):
+        logger.info("Generating %s report for owner %s under %s",
+                    self.name, owner, self.path)
+        # Log all nodes
+        for node in nodes:
+            logger.info(node)
+
+        path = self.config["path"]
+        alias = self.config["alias"]
+
+        header = "last_access_time,size_in_bytes,readable_size,owner,path\n"
+
+        preview_len = min(preview_len, len(nodes))
+        preview = header
+        for node in nodes[0:preview_len]:
+            preview += "%s,%s,%s,%s,%s\n" % (
+                node.subtree_atime,
+                node.subtree_size,
+                bytes2human_readable(node.subtree_size),
+                node.owner,
+                node.path.replace(path, alias, 1))
+        if preview_len < len(nodes):
+            preview += "...\n"
+
+        data = header
+        max_len = min(MAX_NODES_IN_REPORT, len(nodes))
+        for node in nodes[0:max_len]:
+            cur_node = "%s,%s,%s,%s,%s\n" % (
+                node.subtree_atime,
+                node.subtree_size,
+                bytes2human_readable(node.subtree_size),
+                node.owner,
+                node.path.replace(path, alias, 1))
+            data += cur_node
+
+        report = {
+            "filename": "%s_boundary_paths_%s.csv" % (
+                self.name, datetime.fromtimestamp(int(time.time()))),
+            "data": data
+        }
+
+        return preview, report
+
+    def delete_nodes(self):
+        if len(self.nodes) == 0:
+            logger.info("No nodes to delete for rule %s.", self.name)
+            return
+
+        logger.info("Deleting nodes for rule %s...", self.name)
+        for node in self.nodes:
+            if os.path.exists(node.path):
+                if node.isdir:
+                    logger.info("Deleting directory %s...", node.path)
+                    os.rmdir(node.path)
+                else:
+                    logger.info("Deleting file %s...", node.path)
+                    os.remove(node.path)
+            else:
+                logger.warning("%s does not exist", node.path)

```

Comment:
Maybe we should add try/catch in side for loop. In case of, although it may be rare, file is deleted between os.path.exists and os.rmdir

Also, is it possible to reduce the risk of doing os.rmdir ("/*") or something similar.....